subprojects/core/src/integTest/groovy/org/gradle/api/tasks/CopyPermissionsIntegrationTest.groovy

```diff
@@ -222,4 +223,31 @@ class CopyPermissionsIntegrationTest extends AbstractIntegrationSpec {
         cleanup:
         file('src/unauthorized').mode = 0777
     }
+
+    @Issue('https://github.com/gradle/gradle/issues/9576')
+    def "unreadable excluded #type is ignored"() {
+        given:
+        def unreadable = create(file("targetFolder/unreadable${type.capitalize()}"))
+        unreadable.readable = false
+
+        buildFile << """
+            task copy(type: Copy) {
+                from('.') { include '${buildFile.name}' }
```

Comment:
Can we copy another file? Copying the `build.gradle` file doesn't seem like being necessary for the test. I think copying an arbitrary `input.txt` would be better. Then using `from('input.txt')` may also be easier to read.

Reply:
Sure, I didn't like this either, it's what the reporter posted, didn't want to change it too much.
Now I have a good excuse to do it, thanks :)

Reply:
Done